client/listener.ts

```diff
@@ -74,6 +75,36 @@ const handleEvents = async (blockNumber, events, done) => {
       } catch (e) {
         // Likely duplicate
       }
+    } else if(event.name === "LockupCreated") {
+      try {
+        await prisma.lockup.create({
+          data: {
+            address: event.values.provider,
+            amount: event.values.amount,
+            end: event.values.end,
+            weeks: parseInt(((event.values.end - now) / 604800).toFixed()),
```

Comment:
ParseInt is basically doing a Math.floor right? In case the transaction is mined a bit later could this potentially result in a week being just under a natural number?

Meaning that this code: `(event.values.end - now) / 604800` could result in a number like `5.999` and would be rounded down to 5. Or will `event.timestamp` alway correlate to a correct `event.values.end`?

So this is nitpicking a bit, but might be more in line with defensive programming to replace `parseInt` with `Math.round`

Reply:
Good catch and makes sense as it's safer to derive a whole number with `Math.round`.

There's a bigger point here, too:

The final average weeks figure presented will rarely be a whole number (`total weeks / number of lockups`), so we could take the decimal on insert (as opposed to rounding at all).

Though, from a UX perspective, 46.57 weeks (or whatever) is always tougher to grok than 46 weeks or 46 weeks, 4 days.

I vote we go with rounded for now as that's how the inputs present lockup lengths. So I've implemented your suggestion in the listener and the front-end component that displays the average figure.

https://github.com/OriginProtocol/ousd-governance/blob/5cfbb423a31f1e42e138291659eeac8ceb633edf/client/components/vote-escrow/LockupStats.tsx#L28

Reply:
Agree with your suggestion